### DISCUSSION TITLE
Cart Transforms should be visible to other functions

### DISCUSSION THREAD

**Discussion by BrandoCommando** [Shopify/function-examples#264] [Feedback: Cart Transform API]
It appears anything done in a Cart Transform function is not indicated on the input to Discount functions. This can result in double-discounting, which could be avoided by specifying the "cost.totalAmount.amount" as being different than the "cost.subtotalAmount.amount", as the API docs would suggest. However, I have found this not to be the case. Both functions are unaware of each other.

**Comment by BrandoCommando** [comment]
Also, it's called the **Cart** Transform API, but it doesn't change the Cart, it only affects Checkout. Seems to me like it should do both.

**Reply by BrandoCommando** [reply]
When I check out with multiple transforms, the order that's created indicates "Part of" in Shopify Admin, but I don't see any evidence of this in REST/Graphql order resources. It would be nice to have visibility on this at the Order data level.

**Reply by BrandoCommando** [reply]
It looks like "Part of" actually is in the default Order confirmation liquid:
```liquid
          {% for group in line.groups %}
            <span class="order-list__item-variant">Part of: {{ group.display_title }}</span><br/>
          {% endfor %}
```
I can't find "groups" anywhere in the docs though, so maybe that'll come soon.

**Reply by BrandoCommando** [reply]
What would be nice is if price adjustments are done in Transformations, that would be indicate in order.discount_allocations, similar to Scripts.

**Reply by ignacio-chiazzo** [reply]
> When I check out with multiple transforms, the order that's created indicates "Part of" in Shopify Admin, but I don't see any evidence of this in REST/Graphql order resources. It would be nice to have visibility on this at the Order data level.

We are adding APIs under https://shopify.dev/docs/api/admin-graphql/unstable/objects/LineItem

**Reply by ignacio-chiazzo** [reply]
>What would be nice is if price adjustments are done in Transformations, that would be indicate in order.discount_allocations, similar to Scripts.

What do you mean by transformations?

**Reply by BrandoCommando** [reply]
transformations = runs through a Cart Transform function.
With the example Cart Transform output below, I have 3 items that are normally $3.99, but you can bundle 3 for $9.99. The 
If I also have a coupon function run, the input for any Cart Line item Cost shows $3.33 with no indication that a discount has been made.

```javascript
{
  "operations": [
    {
      "merge": {
        "cartLines": [
          { "cartLineId": "gid://shopify/CartLine/420cac91-18b8-4756-88ca-a22db6e43a04", "quantity": 1 },
          { "cartLineId": "gid://shopify/CartLine/f3626945-056c-486a-8b6d-bfbce77b3b86", "quantity": 1 },
          { "cartLineId": "gid://shopify/CartLine/b96328a0-bb87-46ad-b032-39c042ea5999", "quantity": 1 }
        ],
        "parentVariantId": "gid://shopify/ProductVariant/45425466048825",
        "price": {
          "percentageDecrease": {
            "value": 16.541353383458645
          }
        },
        "title": "3-pack for $9.99"
      }
    }
  ]
}
```

**Comment by ignacio-chiazzo** [comment]
> Also, it's called the Cart Transform API, but it doesn't change the Cart, it only affects Checkout. Seems to me like it should do both.

It should be both. If that's not the case for your test store, let me know and I can double-check.

**Reply by BrandoCommando** [reply]
Thanks for the info, @ignacio-chiazzo. I've sent you my test store info.
When I use Scripts, I get line_level_discount_allocations in cart.json:
```javascript
{
  "items": [{
    "original_price": 399,
    "final_price": 333,
    "line_level_discount_allocations": [{
      "amount": 66,
      "discount_application": {
        "type": "script",
        "title": "3-pack for $9.99",
        "value": "0.66",
        "value_type": "fixed_amount",
        "allocation_method": "across",
        "target_selection": "explicit",
        "target_type": "line_item",
        "total_allocated_amount": 66
      }}]}]}
```
It would be nice to see something like this when using Cart Transformation functions, as well.
